lib/ansible/modules/cloud/ovirt/ovirt_vnics_profile.py

```diff
@@ -0,0 +1,254 @@
+from ansible.module_utils.ovirt import (
+    BaseModule,
+    check_sdk,
+    create_connection,
+    equal,
+    get_link_name,
+    ovirt_full_argument_spec,
+    search_by_name,
+    get_id_by_name
+)
+
+
+class EntityVnicPorfileModule(BaseModule):
+
+    def __init__(self, *args, **kwargs):
+        super(EntityVnicPorfileModule, self).__init__(*args, **kwargs)
+
+    def __get_network_id(self):
+        dcs_service = self._connection.system_service().data_centers_service()
+        dc_id = get_id_by_name(dcs_service, self.param('data_center'))
+        networks_service = dcs_service.service(dc_id).networks_service()
+        return get_id_by_name(networks_service, self.param('network'))
+
+    def __get_qos_id(self):
+        dcs_service = self._connection.system_service().data_centers_service()
+        dc_id = get_id_by_name(dcs_service, self.param('data_center'))
+        qoss_service = dcs_service.service(dc_id).qoss_service()
+        return get_id_by_name(qoss_service, self.param('qos'))
+
+    def __get_network_filter_id(self):
+        nf_service = self._connection.system_service().network_filters_service()
+        return get_id_by_name(nf_service, self.param('network_filter'))
+
+    def build_entity(self):
+        return otypes.VnicProfile(
+            name=self.param('name'),
+            network=otypes.Network(id=self.__get_network_id()),
+            description=self.param('description')
+            if self.param('description') else None,
+            port_mirroring=self.param('port_mirroring')
+            if self.param('port_mirroring') is not None else None,
+            pass_through=otypes.VnicPassThrough(mode=otypes.VnicPassThroughMode(self.param('pass_through')))
+            if self.param('pass_through') else None,
+            migratable=self.param('migratable')
+            if self.param('migratable') is not None else None,
+            custom_properties=[
+                otypes.CustomProperty(
+                    name=cp.get('name'),
+                    regexp=cp.get('regexp'),
+                    value=str(cp.get('value')),
+                ) for cp in self.param('custom_properties') if cp
+            ] if self.param('custom_properties') is not None else None,
+            qos=otypes.Qos(id=self.__get_qos_id())
+            if self.param('qos') else None,
+            network_filter=otypes.NetworkFilter(id=self.__get_network_filter_id())
+            if self.param('network_filter') else None
+
+        )
+
+    def update_check(self, entity):
+        return (
+            equal(self.param('migratable'), getattr(entity, 'migratable', None)) and
+            equal(self.param('pass_through'), entity.pass_through.mode) and
+            equal(self.param('description'), entity.description) and
+            equal(self.param('network_filter'), entity.network_filter) and
+            equal(self.param('qos'), entity.qos) and
+            equal(self.param('custom_properties'), entity.custom_properties) and
```

Comment:
Here you are comparing list of dictionaries with list of SDK custom property objects